Add comments about what is in scope for mathlib (#526)
leanprover-community/leanprover-community.github.io: templates/contribute/index.md

   - The explanation of [naming conventions](naming.html).
   - The [documentation guidelines](doc.html).

## What to contribute to mathlib

The first question you will need to consider is whether the material you want to contribute is a good fit for `mathlib`. Whilst there is no formal description of exactly what mathlib's remit is, here are some questions which you can ask about your proposed contribution.

> * Is the material typically covered in an undergraduate or graduate level course in a research mathematics department? If not, then the material may not in scope for `mathlib`.

Reviewer: ```suggestion
* Is the material typically covered in an undergraduate or graduate level course in a research mathematics department? If not, then the material may not be in scope for `mathlib`.
```

   - The explanation of [naming conventions](naming.html).
   - The [documentation guidelines](doc.html).

> ## What to contribute to mathlib

Reviewer 2: Somewhere you should also say: "Discuss on Zulip whether this is a good topic to add"

In particular the remit of mathlib should *not* be thought of as "all of mathematics and related areas". As the number of open PRs increases, the maintainers will sometimes need to make some hard decisions.

An issue related to the fact that the expertise of the maintainers may not cover all of mathematics: you may want to think about *who* is going to review your potential PR. Contributors are encouraged to seek out reviewers for their PRs. A PR reviewer does *not* have to be a maintainer! Contributions from reviewers, especially new reviewers, are essentially always welcome.

If it is not completely clear to you that the code in your project is a good fit for mathlib, you might want to consider creating a standalone repository, and adding `mathlib` as a dependency. There are many Lean repositories on github, indexed by [reservoir](https://reservoir.lean-lang.org). This solution is a particularly good fit for projects in areas which do not align with the expertise of the mathlib maintainers.

## Working on mathlib

> If you believe that your code does belong in mathlib, you should open a PR.

Reviewer 2: This sentence duplicates the first sentence in the section `Making a Pull Request (PR)` and seems more fitting there.

   - The explanation of [naming conventions](naming.html).
   - The [documentation guidelines](doc.html).

## What to contribute to mathlib

Small fixes (for example fixes in docstrings) and single-lemma additions in already-existing theories

> are almost always welcome as contributions to mathlib. Longer PRs which extend existing theories is also almost

Reviewer 3: ```suggestion
are almost always welcome as contributions to mathlib. Longer PRs which extend existing theories are also almost
```

If you are not sure about whether your proposed topic is a good fit for mathlib, then please feel
free to open a discussion in the [`#mathlib` channel](https://leanprover.zulipchat.com/#narrow/channel/287929-mathlib4/) on the Lean Zulip.

An issue related to the fact that the expertise of the maintainers may not cover all of mathematics: 
you may want to think about *who* is going to review your potential PR. Contributors are encouraged 
to seek out reviewers for their PRs. A PR reviewer does *not* have to be a maintainer! This seems
to be a common misconception by the community. Reviews of PRs, especially from new reviewers, 
are essentially always welcome. 

Please also consider the possibility of creating a standalone repository, and adding `mathlib` as a dependency. 
There are many Lean repositories on github, indexed by [reservoir](https://reservoir.lean-lang.org). 
This solution is a particularly good fit for projects in areas which do not align with the 
expertise of the mathlib maintainers. One example of such a repository is the combinatorial game
theory repository. It is also a good fit for projects which would like to move

> quickly; at the time of writing mathlib has over 1000 PRs and it may take time for mathlib

Author: ```suggestion
quickly; at the time of writing mathlib just under 2000 PRs and it may take time for mathlib
```

   - The explanation of [naming conventions](naming.html).
   - The [documentation guidelines](doc.html).

## What to contribute to mathlib

Small fixes (for example fixes in docstrings) and single-lemma additions in already-existing theories

> are almost always welcome as contributions to mathlib. Longer PRs which extend existing theories are also almost

Reviewer 4: We may want to add somethiing here about starting a discussion on Zulip before opening large numbers of PRs (>10?), even if they individually only contain small fixes.

An issue related to the fact that the expertise of the maintainers may not cover all of mathematics: 
you may want to think about *who* is going to review your potential PR. Contributors are encouraged 
to seek out reviewers for their PRs. A PR reviewer does *not* have to be a maintainer! This seems
to be a common misconception by the community. Reviews of PRs, especially from new reviewers, 
are essentially always welcome. 

Please also consider the possibility of creating a standalone repository, and adding `mathlib` as a dependency. 
There are many Lean repositories on github, indexed by [reservoir](https://reservoir.lean-lang.org). 
And [here](https://reservoir.lean-lang.org/@leanprover-community/mathlib/dependents) are those projects
which have `mathlib` has a dependency. The solution of having a new project which depends on
`mathlib` is a particularly good fit for projects in areas which do not align with the 
expertise of the mathlib maintainers. One example of such a repository is the [combinatorial game
theory repository](https://github.com/vihdzp/combinatorial-games). This solution is also a good fit

> for projects which would like to move quickly; at the time of writing mathlib just under 2000 PRs

Reviewer 4: ```suggestion
for projects which would like to move quickly; at the time of writing (Sep 2025), mathlib has just under 2000 PRs
```
